src/v/kafka/server/protocol.cc

```diff
@@ -95,18 +96,47 @@ coordinator_ntp_mapper& protocol::coordinator_mapper() {
 }
 
 config::broker_authn_method get_authn_method(const net::connection& conn) {
+    // If authn_method is set on the endpoint
+    //    Use it
+    // Else if kafka_enable_authorization is not set
+    //    Use sasl if enable_sasl
+    // Else if has mtls mapping rules
+    //    Use mtls_identity
+    // Else
+    //    Disable AuthN
+
+    std::optional<config::broker_authn_method> authn_method;
+    auto n = conn.name();
+    const auto& kafka_api = config::node().kafka_api.value();
+    auto ep_it = std::find_if(
+      kafka_api.begin(),
+      kafka_api.end(),
+      [&n](const config::broker_authn_endpoint& ep) { return ep.name == n; });
+    if (ep_it != kafka_api.end()) {
+        authn_method = ep_it->authn_method;
+    }
+    if (authn_method.has_value()) {
+        return *authn_method;
+    }
     const auto& config = config::shard_local_cfg();
-    if (config.enable_sasl()) {
+    // if kafka_enable_authorization is set, ignore enable_sasl
```

Comment:
Nit: this comment is the other way around: "if kafka_enable_authorization is not set, enable sasl`